#Análisis Exploratorio de Puntaje Crediticio
[Credit Score Classification Cleaned Dataset](https://www.kaggle.com/datasets/iremnurtokuroglu/credit-score-classification-cleaned-dataset/data)

**Puntaje Crediticio**

El puntaje crediticio (credit score) es un número que resume la capacidad de una persona para cumplir con sus obligaciones financieras. Se calcula a partir del historial crediticio de la persona e influye directamente al momento de querer adquirir productos financieros como tarjetas de crédito o préstamos.

Es relevante principalmente para la industria financiera pues les permite calcular riesgos de pérdidas y categorizar clientes.

    En USA el credit score se encuentra entre los 300 y 850 puntos done 850 es la más alta confiabilidad de pago.

**Objetivo del análisis**

Observar cuales sonn los principales factores que afectan en el puntaje crediticio, encontrar similitudes o diferencias entre los clientes que nos ayuden a formar una postura entre las diferentes capacidades financieras comparadas con su puntaje crediticio.

##Cargar el dataset

In [ ]:
#Libreria Kaggle para descargar el dataset
import kagglehub
# Descargar la ultima verion del dataset
path = kagglehub.dataset_download("iremnurtokuroglu/credit-score-classification-cleaned-dataset")
#Muestra la direccion del dataset dentro del notebook
print("Path to dataset files:", path)

Path to dataset files: /kaggle/input/credit-score-classification-cleaned-dataset


In [ ]:
#Importar librerias necesarias
import seaborn as sns
import matplotlib.pyplot as plt
import plotly.express as px
import numpy as np
import pandas as pd
import plotly.graph_objects as go
from plotly.subplots import make_subplots

##Análisis preliminar del dataset

In [ ]:
#Leer el dataset
df = pd.read_csv(path+'/credit_score_cleaned_train.csv')
#mostrar las primeras cinco lineas
df.head()

,id,customer_id,month,name,age,ssn,occupation,annual_income,monthly_inhand_salary,credit_history_age,...,changed_credit_limit,num_credit_inquiries,credit_mix,outstanding_debt,credit_utilization_ratio,payment_of_min_amount,amount_invested_monthly,payment_behaviour,monthly_balance,credit_score
0,0x1602,CUS_0xd40,January,Aaron Maashoh,23,821-00-0265,Scientist,19114.12,1824.8434,89,...,11.27,4,Good,809.98,26.822620,No,80.415300,High_spent_Small_value_payments,312.49408,2
1,0x1603,CUS_0xd40,February,Aaron Maashoh,23,821-00-0265,Scientist,19114.12,1824.8434,93,...,11.27,4,Good,809.98,31.944960,No,118.280220,Low_spent_Large_value_payments,284.62915,2
2,0x1604,CUS_0xd40,March,Aaron Maashoh,23,821-00-0265,Scientist,19114.12,1824.8434,91,...,11.27,4,Good,809.98,28.609352,No,81.699524,Low_spent_Medium_value_payments,331.20987,2
3,0x1605,CUS_0xd40,April,Aaron Maashoh,23,821-00-0265,Scientist,19114.12,1824.8434,92,...,6.27,4,Good,809.98,31.377861,No,199.458070,Low_spent_Small_value_payments,223.45131,2
4,0x1606,CUS_0xd40,May,Aaron Maashoh,23,821-00-0265,Scientist,19114.12,1824.8434,93,...,11.27,4,Good,809.98,24.797346,No,41.420155,High_spent_Medium_value_payments,341.48923,2


In [ ]:
#muestra las ultimas cinco lineas
df.tail()

,id,customer_id,month,name,age,ssn,occupation,annual_income,monthly_inhand_salary,credit_history_age,...,changed_credit_limit,num_credit_inquiries,credit_mix,outstanding_debt,credit_utilization_ratio,payment_of_min_amount,amount_invested_monthly,payment_behaviour,monthly_balance,credit_score
96691,0x25fe9,CUS_0x942c,April,Nicks,25,078-73-5990,Mechanic,39628.99,3359.4158,130,...,11.5,3,Good,502.38,34.663570,No,60.971333,High_spent_Large_value_payments,479.86624,0
96692,0x25fea,CUS_0x942c,May,Nicks,25,078-73-5990,Mechanic,39628.99,3359.4158,131,...,11.5,3,Good,502.38,40.565630,No,54.185950,High_spent_Medium_value_payments,496.65160,0
96693,0x25feb,CUS_0x942c,June,Nicks,25,078-73-5990,Mechanic,39628.99,3359.4158,132,...,11.5,3,Good,502.38,41.255524,No,24.028477,High_spent_Large_value_payments,516.80910,0
96694,0x25fec,CUS_0x942c,July,Nicks,25,078-73-5990,Mechanic,39628.99,3359.4158,133,...,11.5,3,Good,502.38,33.638206,No,251.672580,Low_spent_Large_value_payments,319.16498,1
96695,0x25fed,CUS_0x942c,August,Nicks,25,078-73-5990,Mechanic,39628.99,3359.4158,134,...,11.5,3,Good,502.38,34.192463,No,167.163860,Low_spent_Large_value_payments,393.67370,0


In [ ]:
#Preguntar por la estructura del dataset
df.shape

(96696, 28)

Podemos observar que este dataset cuenta con:

    Filas: 9669628
    Columnas: 28
Esto representa una cantidad considerable de registros que nos ayudara a tener un analisis más estrecho de las variables a analizar

In [ ]:
#Mostrar la informacion del dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 96696 entries, 0 to 96695
Data columns (total 28 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   id                        96696 non-null  object 
 1   customer_id               96696 non-null  object 
 2   month                     96696 non-null  object 
 3   name                      96696 non-null  object 
 4   age                       96696 non-null  int64  
 5   ssn                       96696 non-null  object 
 6   occupation                96696 non-null  object 
 7   annual_income             96696 non-null  float64
 8   monthly_inhand_salary     96696 non-null  float64
 9   credit_history_age        96696 non-null  int64  
 10  total_emi_per_month       96696 non-null  float64
 11  num_bank_accounts         96696 non-null  int64  
 12  num_credit_card           96696 non-null  int64  
 13  interest_rate             96696 non-null  int64  
 14  num_of

In [ ]:
#Mostrar el analisis estadistico del dataset y las variables numericas
df.describe()

,age,annual_income,monthly_inhand_salary,credit_history_age,total_emi_per_month,num_bank_accounts,num_credit_card,interest_rate,num_of_loan,delay_from_due_date,num_of_delayed_payment,changed_credit_limit,num_credit_inquiries,outstanding_debt,credit_utilization_ratio,amount_invested_monthly,monthly_balance,credit_score
count,96696.000000,96696.000000,96696.000000,96696.000000,96696.000000,96696.000000,96696.000000,96696.000000,96696.000000,96696.000000,96696.000000,96696.000000,96696.000000,96696.000000,96696.000000,96696.000000,96696.000000,96696.000000
mean,33.942448,50898.597475,4231.402393,78.471529,106.502641,5.319134,5.498211,14.347646,3.481344,20.806931,13.215180,10.294523,5.667690,1404.464296,32.302757,641.549306,405.811795,0.898889
std,10.377550,38543.773796,3207.836030,33.291894,130.404213,2.597916,2.057083,8.708524,2.435098,14.785004,6.277732,6.775430,3.846715,1148.594411,5.121575,2049.156494,215.518284,0.676131
min,16.000000,7006.035000,303.645420,-1.000000,0.000000,0.000000,0.000000,1.000000,0.000000,-5.000000,0.000000,-6.490000,0.000000,0.230000,20.000000,0.000000,0.007760,0.000000
25%,25.000000,19480.040000,1634.720800,53.000000,28.636501,3.000000,4.000000,7.000000,2.000000,10.000000,9.000000,5.220000,3.000000,555.670000,28.064556,74.984432,271.129822,0.000000
50%,34.000000,37367.340000,3116.120850,78.000000,65.712006,5.000000,5.000000,13.000000,3.000000,18.000000,14.000000,9.310000,5.000000,1142.390000,32.328069,136.521240,339.009780,1.000000
75%,42.000000,72245.760000,6016.623500,106.000000,145.756870,7.000000,7.000000,20.000000,5.000000,28.000000,18.000000,14.640000,8.000000,1891.100000,36.516658,268.072265,475.785608,1.000000
max,56.000000,179987.280000,15204.634000,147.000000,3776.000000,11.000000,11.000000,34.000000,9.000000,67.000000,28.000000,36.970000,17.000000,4998.070000,50.000000,10000.000000,1602.040500,2.000000


##Preparacion y Limpieza de datos

In [ ]:
#Preguntar por si hay registros duplicados
df.duplicated().sum()

np.int64(0)

In [ ]:
#Preguntar si hay valores nulos
df.isnull().sum()

,0
id,0
customer_id,0
month,0
name,0
age,0
ssn,0
occupation,0
annual_income,0
monthly_inhand_salary,0
credit_history_age,0


Como no hay datos duplicados ni datos nulos no tenemos que arreglar el dataset, sin embargo vamos a separar las columnas para identificar las variables numericas y categoricas.

In [ ]:
#ordenamos las columnas segun el tipode datos que continen
df.dtypes.sort_values()

,0
interest_rate,int64
num_credit_inquiries,int64
num_of_delayed_payment,int64
delay_from_due_date,int64
num_of_loan,int64
num_credit_card,int64
num_bank_accounts,int64
credit_history_age,int64
credit_score,int64
age,int64


In [ ]:
#guardamos las columnas de tipo numericas
num_var=df.select_dtypes(include=['int64','float64'])
num_var.head()

,age,annual_income,monthly_inhand_salary,credit_history_age,total_emi_per_month,num_bank_accounts,num_credit_card,interest_rate,num_of_loan,delay_from_due_date,num_of_delayed_payment,changed_credit_limit,num_credit_inquiries,outstanding_debt,credit_utilization_ratio,amount_invested_monthly,monthly_balance,credit_score
0,23,19114.12,1824.8434,89,49.574947,3,4,3,4,3,7,11.27,4,809.98,26.822620,80.415300,312.49408,2
1,23,19114.12,1824.8434,93,49.574947,3,4,3,4,-1,7,11.27,4,809.98,31.944960,118.280220,284.62915,2
2,23,19114.12,1824.8434,91,49.574947,3,4,3,4,3,7,11.27,4,809.98,28.609352,81.699524,331.20987,2
3,23,19114.12,1824.8434,92,49.574947,3,4,3,4,5,4,6.27,4,809.98,31.377861,199.458070,223.45131,2
4,23,19114.12,1824.8434,93,49.574947,3,4,3,4,6,4,11.27,4,809.98,24.797346,41.420155,341.48923,2


In [ ]:
#Hacemos una copia de las columnas categoricas
cat_var=df.select_dtypes(include=['object'])
cat_var.head()

,id,customer_id,month,name,ssn,occupation,type_of_loan,credit_mix,payment_of_min_amount,payment_behaviour
0,0x1602,CUS_0xd40,January,Aaron Maashoh,821-00-0265,Scientist,"['Auto Loan', 'Credit-Builder Loan', 'Personal...",Good,No,High_spent_Small_value_payments
1,0x1603,CUS_0xd40,February,Aaron Maashoh,821-00-0265,Scientist,"['Auto Loan', 'Credit-Builder Loan', 'Personal...",Good,No,Low_spent_Large_value_payments
2,0x1604,CUS_0xd40,March,Aaron Maashoh,821-00-0265,Scientist,"['Auto Loan', 'Credit-Builder Loan', 'Personal...",Good,No,Low_spent_Medium_value_payments
3,0x1605,CUS_0xd40,April,Aaron Maashoh,821-00-0265,Scientist,"['Auto Loan', 'Credit-Builder Loan', 'Personal...",Good,No,Low_spent_Small_value_payments
4,0x1606,CUS_0xd40,May,Aaron Maashoh,821-00-0265,Scientist,"['Auto Loan', 'Credit-Builder Loan', 'Personal...",Good,No,High_spent_Medium_value_payments


In [ ]:
#Eliminamos las columnas de id, nombre y numero de seguro social
cat_var.drop(['id','name','ssn'], axis=1, inplace=True)
cat_var.head()

,customer_id,month,occupation,type_of_loan,credit_mix,payment_of_min_amount,payment_behaviour
0,CUS_0xd40,January,Scientist,"['Auto Loan', 'Credit-Builder Loan', 'Personal...",Good,No,High_spent_Small_value_payments
1,CUS_0xd40,February,Scientist,"['Auto Loan', 'Credit-Builder Loan', 'Personal...",Good,No,Low_spent_Large_value_payments
2,CUS_0xd40,March,Scientist,"['Auto Loan', 'Credit-Builder Loan', 'Personal...",Good,No,Low_spent_Medium_value_payments
3,CUS_0xd40,April,Scientist,"['Auto Loan', 'Credit-Builder Loan', 'Personal...",Good,No,Low_spent_Small_value_payments
4,CUS_0xd40,May,Scientist,"['Auto Loan', 'Credit-Builder Loan', 'Personal...",Good,No,High_spent_Medium_value_payments


Eliminamos las columnas de nombre y numero de seguro social porque son datos que requieren un trato especial debido a que se trata de información sensible, además de que no son datos relevantes para este analisis.

##Análisis y Visualización de Datos

In [ ]:
#guardamos la relacion de una columna con otra
Relacion_var=num_var.corr()
Relacion_var

,age,annual_income,monthly_inhand_salary,credit_history_age,total_emi_per_month,num_bank_accounts,num_credit_card,interest_rate,num_of_loan,delay_from_due_date,num_of_delayed_payment,changed_credit_limit,num_credit_inquiries,outstanding_debt,credit_utilization_ratio,amount_invested_monthly,monthly_balance,credit_score
age,1.000000,0.078625,0.077739,0.209324,-0.044582,-0.168559,-0.128250,-0.195105,-0.190940,-0.155064,-0.162070,-0.142125,-0.219740,-0.183260,0.020972,0.002886,0.101518,0.142782
annual_income,0.078625,1.000000,0.998214,0.268752,0.446004,-0.282102,-0.214969,-0.300181,-0.253808,-0.249060,-0.286300,-0.177214,-0.279408,-0.267425,0.177664,0.059337,0.705753,0.209451
monthly_inhand_salary,0.077739,0.998214,1.000000,0.267791,0.429779,-0.281247,-0.213953,-0.299698,-0.252501,-0.248395,-0.285378,-0.177161,-0.278991,-0.267111,0.177765,0.059387,0.706901,0.206809
credit_history_age,0.209324,0.268752,0.267791,1.000000,-0.112512,-0.481399,-0.413966,-0.574777,-0.602823,-0.490685,-0.479022,-0.425350,-0.609140,-0.626612,0.071281,0.018174,0.320314,0.386226
total_emi_per_month,-0.044582,0.446004,0.429779,-0.112512,1.000000,0.049116,0.058835,0.072229,0.331579,0.080608,0.051075,0.063297,0.101042,0.110004,0.024192,0.028511,0.097034,0.016795
num_bank_accounts,-0.168559,-0.282102,-0.281247,-0.481399,0.049116,1.000000,0.439339,0.584901,0.470607,0.561154,0.601852,0.333795,0.517098,0.505493,-0.071750,-0.020056,-0.297621,-0.385838
num_credit_card,-0.128250,-0.214969,-0.213953,-0.413966,0.058835,0.439339,1.000000,0.494626,0.418095,0.476670,0.423034,0.256119,0.457133,0.487489,-0.054875,-0.012259,-0.236132,-0.402888
interest_rate,-0.195105,-0.300181,-0.299698,-0.574777,0.072229,0.584901,0.494626,1.000000,0.560564,0.588343,0.571130,0.373258,0.633720,0.629029,-0.075431,-0.017756,-0.326391,-0.483274
num_of_loan,-0.190940,-0.253808,-0.252501,-0.602823,0.331579,0.470607,0.418095,0.560564,1.000000,0.503128,0.472061,0.373724,0.568274,0.638909,-0.100823,-0.014366,-0.431056,-0.359013
delay_from_due_date,-0.155064,-0.249060,-0.248395,-0.490685,0.080608,0.561154,0.476670,0.588343,0.503128,1.000000,0.541987,0.298827,0.540173,0.573000,-0.062715,-0.011838,-0.280576,-0.431728


In [ ]:
#al hablar de relacion los datos son entre 0 - 1 por lo tanto no necesitaamo normalizarlo
#creamos una figura a la cual asignaremos un mapa de calor con los datos anteriores
fig = px.imshow(Relacion_var, aspect="auto",color_continuous_scale="Viridis")
fig.show()

Contamos con 18 variables numericas, por tanto decidimos preguntar por su relación entre variables, esto nos ayuda a identificar las columnas que podrian daarnos información de los datos, nos vamos a concentrar en las variables que se encuentran en los extremos.

Considerando la grafica anterior las variables con mayor relación (positiva o negativa) al puntaje crediticio son:
* Historial crediticio en meses (credit_history_age)
* Número de tarjetas de credito (num_credit_card)
* Tasa de interés (interest_rate)
* Número de prestamos (num_of_loan)
* Número de pagos retrasados (num_of_delayed_payment)
* Número de consultas de credito (num_credit_inquiries)
* Deuda pendiente (outstanding_debt)
* Balance mensual(monthly_balance)

Sin embargo, consideramos que las siguiente variables nos dan informacion acerca de los clientes por lo que decidimos considerarlas:
* Edad (age)
* Ingresos Anuales (annual_income)

Para hacer el análisis decidimos visualizar los datos en 2 categorias, según la edad y la ocupación de los clientes.

In [ ]:
#creamos una variables en la cual evaluaremos segun el cliente por tanto creamos una copia ordenada de las columnas numericas
bycustomer=num_var.sort_values(by='credit_score', ascending=False)
#agregamos el id del cliente que nos va a permitir agrupar los datos
bycustomer['customer_id']=df['customer_id']
#agrupamos segun el cliente promediando las coincidencias
bycustomer=bycustomer.groupby('customer_id').mean()
#convertimos eel indice a columna
bycustomer.reset_index( inplace=True)
#creamos una figura en la cual agregaremos un grafico que relacione la edad con el historial crediticiio
fig = px.scatter(bycustomer,x="age", y="credit_history_age",color="credit_history_age",color_continuous_scale="Cividis",
                 title="Relación entre Edad y el Historial Crediticio",hover_data=["credit_score","customer_id"],trendline="ols")
#cambiamos el tamaño de los ciculitos (marcas)
fig.update_traces(marker=dict(size=5))
#guardamos el id del outlier en una variables
outlierid = "CUS_0x7aba"
#buscamos el registro con ese id
outlierdata = bycustomer[bycustomer['customer_id'] == outlierid].iloc[0]
#agregamos anotaciones para señalar el outlier con unaa flecha y que indique el ID
fig.add_annotation(x=outlierdata['age'],y=outlierdata['credit_history_age'],text=f"Outlier ID: {outlierid}",arrowhead=2,
    arrowsize=1,arrowwidth=1.5,arrowcolor="black")
fig.show()

Podemos ver una marcada relacion entre la edad y el historial crediticio pues despues de los 46 años encontramos una antiguedad mayor a 60 meses (mayor a 5 años).

Sin embargo tambien encontramos un outlier puesto que el cliente cuenta con un historial de 137 meses (más de 11 años) lo que suena un poco ironico considerando que el cliente tiene 16 años, significa que el cliente comenzo con su historial desde los 5 años por tanto podemos considerar que se trata de un error.

In [ ]:
#mostramos el registro del outlier
outlierdata

,6892
customer_id,CUS_0x7aba
age,16.0
annual_income,31136.8
monthly_inhand_salary,2612.7334
credit_history_age,135.75
total_emi_per_month,88.99986
num_bank_accounts,4.0
num_credit_card,7.0
interest_rate,18.0
num_of_loan,4.0


En general este outlier tiene no tiene un comportamiento irregular, más que su relacion entre su edad y historial crediticio como mencionabamos anteriormente puede tratarse de un error de registro.

###Edad

In [ ]:
#creamos un dataframe que cree una copia de las columnas numericas ordenada por puntaje crediticio
byage=num_var.sort_values(by='credit_score', ascending=False)
#agrupamos segun la edad
byage=byage.groupby('age').mean()
byage.head()

,annual_income,monthly_inhand_salary,credit_history_age,total_emi_per_month,num_bank_accounts,num_credit_card,interest_rate,num_of_loan,delay_from_due_date,num_of_delayed_payment,changed_credit_limit,num_credit_inquiries,outstanding_debt,credit_utilization_ratio,amount_invested_monthly,monthly_balance,credit_score
age,,,,,,,,,,,,,,,,,
16,41213.300926,3386.752300,54.019608,150.752137,6.607843,5.994118,20.031373,5.159804,28.859804,16.673529,13.500471,8.450000,1976.954010,31.709195,576.043122,315.036013,0.602941
17,35358.486614,2934.832430,53.221507,114.274299,6.698648,6.365744,20.390856,4.965873,28.934321,16.986478,13.904604,8.625241,2118.528055,31.946384,596.392154,320.422947,0.585963
18,48533.912330,4047.480325,70.493650,116.685315,5.701762,5.785744,15.842687,4.058583,23.602212,14.453912,11.287603,6.370340,1706.291237,32.060048,622.929640,381.481480,0.818107
19,50646.064425,4200.865742,73.713589,115.864866,5.534495,5.613240,15.481882,3.770383,22.558885,13.811847,10.715338,6.105923,1563.688578,32.229009,711.766645,398.099268,0.831359
20,52394.730157,4367.679632,77.877688,119.331836,5.471977,5.566091,14.984491,3.631653,21.146986,13.408178,10.742284,6.188579,1513.226916,32.364498,698.089441,403.847401,0.896017


In [ ]:
#declaramos en una lista las columnas que queremos analizar
cols = ['annual_income','num_credit_card','interest_rate','num_of_loan','num_of_delayed_payment',
        'num_credit_inquiries','outstanding_debt','monthly_balance','credit_score']
byage.reset_index(inplace=True)
#creamos una figura como cuadricula 3*3 donde agreguemos cada grafica y cambiaremos los _ en el nombre de las columnas
fig = make_subplots(rows=3, cols=3,subplot_titles=[col.replace("_", " ").title() for col in cols])
#inicializamos las columnas y filas en 1
row = 1
col = 1
#creamos un ciclo que repita el grafico el numero de columnas
for i, column in enumerate(cols):
  #agregamos el grafico a la figura
  fig.add_trace(
      #asignamos el grafico la columna y la edad
      go.Scatter(x=byage['age'], y=byage[column],mode='markers',marker=dict(color=byage['age'],
                  colorscale='Turbo'),name=column),row=row, col=col)
  #sumamos uno a la columna despues de cada grafica
  col += 1
  #cuando tengamos las 3 columnas se asignara otravez el valor a 1 pero se le agregara una fila
  if col > 3:
      col = 1
      row += 1
#cambiamos las dimensiones de presentacion
fig.update_layout(showlegend=False)
fig.show()

####Conclusiones de los gráficos por edad:
* **Ingreso Anual** (annual_income). Podemos ver que hay una estrecha relación entre la edad y los ingresos anuales, encontrando que entre mayor edad mayor es el ingreso.

* **Numero de tarjetas de credito** (num_credit_card). Contrario a los ingresos entre más jovenes sean los clientes tienen una mayor cantidad de tarjetas de credito podemos considerar que al contar con menores ingresos requieren de instrumentos crediticios como tarjetaas para manejar sus ingresos o a la intención de crear un historial crediticio.
* **Tasa de intéreses** (interest_rate). Tambien notamos que la tasa de intéreses se reduce con la edad, que puede tener relación a sus ingresos o la madurez de manejo de herramientas crediticias, por tanto los bancos tienen mayor confianza en estos clientes.
* **Número de prestamos** (num_of_loan). Tenemos un comportamiento parecido a la cantidad de tarjetas de crédito, por lo que apoya nuestra teoría de que como la edad afecta a su uso.
* **Número de pagos pendientes** (num_of_delayed_payment). Reafirmando lo anterior las personas con mayor edaad suelen cumplir con sus pagos de una forma considerablemente mayor a las personas jovenes.
* **Numero de solicitudes de crédito** (num_credit_inquiries). Aqui mantenemos un mayor movimiento entre las personas jovenes, por tanto podemos considerar esta variable como datos que refuerzan al número de préstamos, tambien podemos considerar una posible brecha de edad en el uso de herramientas digitales para consulta de crédito..
* **Deuda Pendiente** (outstanding_debt). Para reafirmar la variable de número de pagos pendientes, las personas más jovenes tienen mayor dificultad para pagar, que puede deberse a su bajo ingreso.
* **Balance mensual** (monthly_balance). Las personas de edad avanzada, cuentan con números más altos en su balance mensual, puede ser a razón de sus ingresos mensuales, y consideraando que son los que menos préstamos tienen, lo que lo hace congruente.
* **Puntaje Crediticio** (credit_score). Finalmente, el propósito de nuestro estudio se ve reflejado en esta grafica por edad, obtenemos que entre mayor edad el puntaje crediticio tambien aumenta, algunas de las varibales pueden ser los ingresos, los tipos de gastos, la confiaabilidad con la banca, el historial crediticio.

Tambien observamos que hay dos puntos críticos, las personas que llegan a los 20 años comienzan a tener un comportamiento similar en todas las graficas, y las personas despúes de los 45 años tienen aparentemente una mayor estabilidad economica.

In [ ]:
#guardamos la desviacion estandar en cada columna
stdage=byage.std()
stdage

,0
index,11.979149
age,11.979149
annual_income,6610.004591
monthly_inhand_salary,547.566138
credit_history_age,13.062154
total_emi_per_month,12.342722
num_bank_accounts,0.792776
num_credit_card,0.485129
interest_rate,3.146925
num_of_loan,0.843515


Podemos notar que las variables con mayor desviación, son la edad con 11 años, el historial crediticio con 13 años, los ingresos con casi $$7000, y las deudas con $370.

###Ocupacion

In [ ]:
#creamos una copia que de las columnas numericas ordenadas por el puntaje crediticio
byoccupation=num_var.sort_values(by='credit_score', ascending=False)
#agregamos la comuna de ocupacion
byoccupation['occupation']=cat_var['occupation']
#agrupamos por ocupacion
byoccupation=byoccupation.groupby('occupation').mean()
byoccupation.head()

,age,annual_income,monthly_inhand_salary,credit_history_age,total_emi_per_month,num_bank_accounts,num_credit_card,interest_rate,num_of_loan,delay_from_due_date,num_of_delayed_payment,changed_credit_limit,num_credit_inquiries,outstanding_debt,credit_utilization_ratio,amount_invested_monthly,monthly_balance,credit_score
occupation,,,,,,,,,,,,,,,,,,
Accountant,34.216564,51264.675944,4263.600829,78.250460,110.593784,5.237577,5.491411,14.273620,3.466258,20.611963,13.261656,10.722724,5.674233,1404.285669,32.317475,602.814378,406.745110,0.891564
Architect,33.955024,52093.902158,4336.364967,80.311522,106.796277,5.297984,5.423135,13.827918,3.371841,20.231498,13.178099,10.017347,5.454874,1310.642046,32.380331,686.413049,413.912957,0.920126
Developer,33.806304,50397.537621,4187.966658,77.804468,103.595396,5.320838,5.606793,14.574051,3.554468,21.120563,13.420135,10.479425,5.804774,1421.628384,32.278770,608.408960,405.468436,0.897032
Doctor,33.684981,49698.211554,4138.015726,78.300693,108.577761,5.330762,5.480793,14.327456,3.644836,20.573835,13.332494,10.269956,5.638224,1425.500529,32.224679,615.574138,400.540581,0.910737
Engineer,33.184855,51377.187372,4272.049166,77.231560,104.649853,5.318319,5.503779,14.455865,3.489722,21.338271,13.139510,10.537189,5.759371,1425.219081,32.434878,646.008494,409.375491,0.887999


In [ ]:
#determinamos cuales son las columnass que vamos a analizar
cols = ['annual_income','interest_rate','num_of_loan','num_credit_inquiries','outstanding_debt','credit_score']
#creamos una figura con cuadricula 2*3 donde agreguemos cada grafica y cambiaremos los _ en el nombre de las columnas
fig = make_subplots(rows=2, cols=3,subplot_titles=[col.replace("_", " ").title() for col in cols])
#inicializamos las columnas y filas a 1
row = 1
col = 1
#creamos un ciclo que repita el proceso el numero de columnass
for i, column in enumerate(cols):
  #agregamos el grafico a la figura
    fig.add_trace(
        #creamos el grafico de barras con la ocupacion y la columna
        go.Bar(x=byoccupation['occupation'], y=byoccupation[column],name=column,
               ),row=row, col=col)
    #agregamos 1 a la columna despues de cada grafico creado
    col += 1
    #cuando las columnas sean mayor a 2 se vuelven a acomodar a 1 pero se cambia de fila
    if col > 3:
        col = 1
        row += 1
#cambiamos las dimensiones de la figura
fig.update_layout(height=800,width=1300,showlegend=False)
fig.show()

####Conclusiones de los gráficos por Ocupación:
* **Ingreso Anual** (annual_income). Podemos destacar que los clientes con ocupación de arquitectos son los que tienen mayor ingresos anuales y los Periodistas son quienes perciben menores ingresos anuales.
* **Tasa de intéreses** (interest_rate). Tambien notamos que la tasa de intéreses más baja en relación a la anterior afirmación es de los arquitectos y la más alta es para los abogados, esto nos crea una condicionante pues apesar de que los abogados no se encuentra como ocupacion con ingresos mas bajos, puede ser a razón de incumplimientos de pagos.
* **Número de prestamos** (num_of_loan). Nuevamente los arquitectos son la ocupación que menos préstamos pide y los doctores son quienes más cantidad de préstamos tienen, lo que nos crea una condicionante de en que gastan los médicos.
* **Numero de solicitudes de crédito** (num_credit_inquiries). Aquí presentamos un cambio de profesión siendo los músicos y mecanicos, quienes tienen menos y más solicitudes de crédito (respectivamente).
* **Deuda Pendiente** (outstanding_debt). Vuelve a protagonizar la ocupación de aarquitecto con menor deuda y mecanico con mayor deuda a una diferencia de casi $300.
* **Puntaje Crediticio** (credit_score). Finalmente, los Media Managers son la ocupación con mayor puntaje crediticio y en el otro extremo los Escritores, que puede deberse a la naturaleza de este trabajo, pues hay muchos picos en la carrera de un escritor.

###Mes

In [ ]:
#creamos una copia de las columnas numericas
bymonth=num_var.sort_values(by='credit_score', ascending=False)
#agregamos la columna de mes
bymonth['month']=cat_var['month']
#agrupamos segun el mes
bymonth=bymonth.groupby('month').mean()
bymonth

,age,annual_income,monthly_inhand_salary,credit_history_age,total_emi_per_month,num_bank_accounts,num_credit_card,interest_rate,num_of_loan,delay_from_due_date,num_of_delayed_payment,changed_credit_limit,num_credit_inquiries,outstanding_debt,credit_utilization_ratio,amount_invested_monthly,monthly_balance,credit_score
month,,,,,,,,,,,,,,,,,,
April,33.896087,50898.597475,4233.133719,78.295441,104.871502,5.320179,5.499297,14.347646,3.481344,20.809299,13.209150,10.291989,5.533631,1404.464296,32.246731,644.521619,406.275111,0.904691
August,34.233722,50898.597475,4221.137105,79.865062,116.646102,5.319269,5.498304,14.347646,3.481344,20.795814,13.207082,10.288514,6.523703,1404.464296,32.363347,638.057213,406.620393,0.913626
February,33.736659,50898.597475,4239.000885,77.796393,99.056484,5.318524,5.497725,14.347646,3.481344,20.808886,13.212377,10.314327,5.053611,1404.464296,32.272119,632.481118,405.187803,0.890544
January,33.657814,50898.597475,4241.272197,76.982626,96.402701,5.317697,5.496980,14.347646,3.481344,20.787540,13.208819,10.313665,4.829155,1404.464296,32.305204,648.408553,408.809553,0.883677
July,34.149086,50898.597475,4223.684817,79.620501,113.824365,5.319020,5.497973,14.347646,3.481344,20.830148,13.231819,10.314492,6.283362,1404.464296,32.277864,635.007198,404.848147,0.912799
June,34.065690,50898.597475,4226.526632,78.675602,111.352318,5.319517,5.498304,14.347646,3.481344,20.800695,13.213122,10.269072,6.038306,1404.464296,32.315910,641.066933,405.264965,0.900554
March,33.818896,50898.597475,4236.412358,78.053942,101.592996,5.319351,5.498469,14.347646,3.481344,20.811616,13.229999,10.276931,5.295028,1404.464296,32.313979,655.906194,405.733999,0.887400
May,33.981633,50898.597475,4230.051434,78.482667,108.274658,5.319517,5.498635,14.347646,3.481344,20.811450,13.209068,10.287190,5.784727,1404.464296,32.326902,636.945624,403.754389,0.897824


In [ ]:
#designamos las columnas que queremos analizar
columnasmes=['total_emi_per_month','num_credit_inquiries']
bymonth.reset_index(inplace=True)
#creamos una figura donde agregaremos los graaficos en una cuadricula de 1*2
fig = make_subplots(rows=1, cols=2,subplot_titles=[col.replace("_", " ").title() for col in columnasmes])
#Inicializamos las columnas a 1
row = 1
col = 1
#creamos un ciclo que repita el proceso el numero de columnas
for i, column in enumerate(columnasmes):
  #agregamos a la figura el grafico
    fig.add_trace(
        #creamos un grafico de barras con el mes y la columna
        go.Bar(y=bymonth['month'], x=bymonth[column],name=column,orientation="h"),row=row, col=col)
    #agregamos a 1 la columna
    col += 1
#cambiamos la dimension de figura
fig.update_layout(showlegend=False)
fig.show()

* **EMI por mes**

        EMI(Equated Monthly Installment). Es una tarifa dispuesta por el banco que esta en relación al préstamo o crédito y que tiene que ser saldada mensualmente.

Cuando analizamos los registros mensualmente podemos obtener que Agosto es el mes con mayor EMI, que puede asociarse a variables sociodemográficas puesto que coincide con el regreso escolar,  ademas de que se encuentra a mitad de año a difrencia de Enero que es el mes con menor EMI, que al igual puede asociarse a variaables como prestaciones y aguinaldos.
* **Número de solicitudes de crédito**

Podemos notar que en relación con el análisis anterior el mes con mayor solicitudes es Agosto y Enero el mes con menor solicitudes.

In [ ]:
#creamos un dataframe para guardar el typo de comportamiento
bypayment=pd.DataFrame()
#agregamos el comportamiento de pago
bypayment['payment_behaviour']=cat_var['payment_behaviour']
#Agregamos el puntaje crediticio
bypayment['credit_score']=num_var['credit_score']
#agrupamos segun el comportamiento
bypayment = bypayment.groupby('payment_behaviour').mean()
#ordenamos segun el puntaje crediticio
bypayment = bypayment.sort_values(by='credit_score', ascending=False)
bypayment

,credit_score
payment_behaviour,
High_spent_Large_value_payments,1.019463
High_spent_Medium_value_payments,0.955673
High_spent_Small_value_payments,0.922110
Low_spent_Large_value_payments,0.907009
Low_spent_Medium_value_payments,0.888812
Low_spent_Small_value_payments,0.785496


###Comportamiento

In [ ]:
bypayment.reset_index(inplace=True)
fig = px.bar(bypayment, x='payment_behaviour', y='credit_score',
             color='credit_score')
fig.show()


El comportamiento de pago es de las variables más importantes para el puntaje crediticio, lo podemos ver claro en esta gráfica, además nos otorga dos características clave para formar un buen puntaje crediticio.

Podemos notar que entre más gastos mayor es el puntaje y que aunque se tengan pagos pequeños tiene mayor puntaje que aquellos que gastan poco y pagan cantidades altas.
Esto nos declara que los puntajes crediticios más altos son de clientes que no solo pagan si no que tambien gastan bastante.

Con lo anterior en mente, nos gustaría cuestionar a que tipo de préstamos se les atribuye el mayor puntaje crediticio.

In [ ]:
typeloan = pd.DataFrame()
typeloan['type_of_loan'] = cat_var['type_of_loan'].str.strip("[]").str.split(", ")
typeloan['credit_score'] = num_var['credit_score']
 #Descompone la lista en filas
typeloan = typeloan.explode('type_of_loan')
# Elimina espacios
typeloan['type_of_loan'] = typeloan['type_of_loan'].str.strip()
typeloan['loan_counts'] = typeloan['type_of_loan'].map(typeloan['type_of_loan'].value_counts())
# Agrupa por tipo de préstamo y calcula promedio
typeloan = typeloan.groupby('type_of_loan').mean()
typeloan

,credit_score,loan_counts
type_of_loan,,
'Auto Loan',0.725528,35960.0
'Credit-Builder Loan',0.734320,38520.0
'Debt Consolidation Loan',0.720685,37216.0
'Home Equity Loan',0.732969,37168.0
'Mortgage Loan',0.728622,37000.0
'No Loan',1.187500,11312.0
'Not Specified',0.737771,37616.0
'Payday Loan',0.726097,38824.0
'Personal Loan',0.729593,37144.0


In [ ]:
typeloan.sort_values(by='credit_score', ascending=False, inplace=True)
fig = px.bar(typeloan, x='type_of_loan', y='credit_score',
             color='type_of_loan',
             title='Puntaje Crediticio Promedio por Tipo de Préstamo')
fig.show()

A pesar de que no existe una gran diferencia notamos que aquellos registros que no tenían préstamos o no eran especificados eran los que contaban con mayor puntaje crediticio, sin embargo no podemos tener una conclusión concreta de esto puesto que estas eran las respuestas con mayor repetición.

##Conclusiones






Este análisis nos permite comprender cuales son las principales variables consideradas en un puntaje créditicio (ingreso anual, la cantidad de préstamos, el historial de crédito y la deuda pendiente), también permiten construir un sentido conscient de lo que representa este número.

> Puntaje Crediticio (credit_score). Finalmente, el propósito de nuestro estudio se ve reflejado en esta grafica por edad, obtenemos que entre mayor edad el puntaje crediticio tambien aumenta, algunas de las varibales pueden ser los ingresos, los tipos de gastos, la confiaabilidad con la banca, el historial crediticio.

Luego de su estudio, se identificaron patrones claros y relevantes entre clientes de alto gasto con pagos grandes y puntajes más altos, en contraste con clientes de bajo gasto y menor cumplimiento, quese resumirán en esta sección.
* Encontramos un impacto en la relación de edad con el historial crediticio, la edad aunque no es una de las variables que afecte al puntaje crediticio tiene relación con otras que sí.

> Podemos ver una marcada relacion entre la edad y el historial crediticio pues despues de los 46 años encontramos una antiguedad mayor a 60 meses (mayor a 5 años).
* Siguiendo con lo anterior la edad refleja comportamientos como inestabilidad financiera antes de los 20, luego se encuentra un equilibrio y cercas de los 45 vuelve a haber un cambio a una probable generosa estabilidad financiera.

> Tambien observamos que hay dos puntos críticos, las personas que llegan a los 20 años comienzan a tener un comportamiento similar en todas las graficas, y las personas despúes de los 45 años tienen aparentemente una mayor estabilidad economica.

* En general vemos que variables como los ingresos y el balance mensual tienen una relación proporcional al puntaje crediticio y la edad.

> Ingreso Anual (annual_income). Podemos ver que hay una estrecha relación entre la edad y los ingresos anuales, encontrando que entre mayor edad mayor es el ingreso.

* En cambio varibles, como las tarjetas de crédito, préstamos, deudas, y solicitudes tienen una relación inversamente proporcional al puntaje crediticio.

> Numero de tarjetas de credito (num_credit_card). Contrario a los ingresos entre más jovenes sean los clientes tienen una mayor cantidad de tarjetas de credito podemos considerar que al contar con menores ingresos requieren de instrumentos crediticios como tarjetaas para manejar sus ingresos o a la intención de crear un historial crediticio.

> Número de pagos pendientes (num_of_delayed_payment). Reafirmando lo anterior las personas con mayor edaad suelen cumplir con sus pagos de una forma considerablemente mayor a las personas jovenes.

* Al analizar ocupaciones, obtenemos que los Media Managers son los que cuentan con el mayor puntaje crediticio, en cambio los escritores son la ocupación con menor puntaje.

> Puntaje Crediticio (credit_score). Finalmente, los Media Managers son la ocupación con mayor puntaje crediticio y en el otro extremo los Escritores, que puede deberse a la naturaleza de este trabajo, pues hay muchos picos en la carrera de un escritor.

* En cuanto a los Media Managers al analizar el comportamiento encontramos que entre mayor gastos en conjunto a una capacidad de costearlos, mas alto es el puntaje por lo que podemos asociarlo a esto.

>Esto nos declara que los puntajes crediticios más altos son de clientes que no solo pagan si no que tambien gastan bastante.


Finalmente, es fácil determinar que estas variables hablan más de lo que una persona puede decir de su capacidad financieraa por tanto el conjunto de estas variables puede ayudar a clasificar clientes y detectar perfiles de riesgo, también es una oportunidad de crear modelos predictivos de riesgo que usen estas mismas variables.



---



    Información del conjunto de datos

    Nombre: Credit Score Classification Cleaned Dataset

    Link: https://www.kaggle.com/datasets/iremnurtokuroglu/credit-score-classification-cleaned-dataset/data
Santillan Nuñez Ana Paula - 246102